pkg/resolver/related.go

```diff
@@ -22,3 +31,274 @@ type SearchRelatedResult struct {
 // 	klog.Info("TODO: Resolve SearchRelatedResult: Items() - model/related.go")
 // 	return nil
 // }
+func (s *SearchResult) buildRelationsQuery() {
+	var whereDs []exp.Expression
+
+	//The level can be parameterized later, if needed
+	whereDs = append(whereDs, goqu.C("level").Lt(4))       // Add filter to select only upto level 4 relationships
+	whereDs = append(whereDs, goqu.C("iid").NotIn(s.uids)) // Add filter to avoid selecting the search object itself
+
+	//Non-recursive term SELECT CLAUSE
+	schema := goqu.S("search")
+	selectBase := make([]interface{}, 0)
+	selectBase = append(selectBase, goqu.L("1").As("level"), "sourceid", "destid", "sourcekind", "destkind", "cluster")
+
+	//Recursive term SELECT CLAUSE
+	selectNext := make([]interface{}, 0)
+	selectNext = append(selectNext, goqu.L("level+1").As("level"), "e.sourceid", "e.destid", "e.sourcekind",
+		"e.destkind", "e.cluster")
+
+	//Combine both source and dest ids and source and dest kinds into one column using UNNEST function
+	selectCombineIds := make([]interface{}, 0)
+	selectCombineIds = append(selectCombineIds, goqu.C("level"),
+		goqu.L("unnest(array[sourceid, destid, concat('cluster__',cluster)])").As("iid"),
+		goqu.L("unnest(array[sourcekind, destkind, 'Cluster'])").As("kind"))
+
+	//Final select statement
+	selectFinal := make([]interface{}, 0)
+	selectFinal = append(selectFinal, goqu.C("iid"), goqu.C("kind"), goqu.MIN("level").As("level"))
+
+	//GROUPBY CLAUSE
+	groupBy := make([]interface{}, 0)
+	groupBy = append(groupBy, goqu.C("iid"), goqu.C("kind"))
+
+	// Original query to find relations between resources - accepts an array of uids
```

Comment:
I don't think we need the `Original` in `Original query...` anymore - this is just a part of current the query.